pkg/cri/sbserver/container_create.go

```diff
@@ -355,11 +366,20 @@ func (c *criService) volumeMounts(containerRootDir string, criMounts []*runtime.
 		}
 		volumeID := util.GenerateID()
 		src := filepath.Join(containerRootDir, "volumes", volumeID)
+
+		// Destination must be an absolute path.
+		// If it isn't, let's make it relative to "/".
+		if !filepath.IsAbs(dst) {
```

Comment:
I don't think we should add this change here. `dst` is given by kubelet.

Reply:
If dst is not an absolute path, runc will throw an error. Before it was a warning.

I'm planning to join containerd community meeting and discuss if we want to ask runc to postpone upgrading the warning to an error. I think it is safer to make it an abs path if it isn't, but I'm not sure how to do that correctly on windows and maybe the kubelet always sends abs paths?

I will try if that get rids of the windows errors, otherwise I'll add a `goruntime.OS != "windows"` and see if that makes the trick. If you have better ideas, they are more than welcome :)

Reply:
Not needed, yay! :)

Reply:
(see the other comment and PR description for why changing this file doesn't seem necessary :))

Reply:
> NOTE: Keep in the timeline for other reviewer.

The kubelet will ensure the path is abs in linux platform. FYI.
 https://github.com/kubernetes/kubernetes/blob/98358b8ce11b0c1878ae7aa1482668cb7a0b0e23/pkg/kubelet/kubelet_pods.go#L249

Reply:
I will try to join the meeting tomorrow night. I'm not sure that we should change relative to abs. Since the code is right here for years, and everything is guaranteed by caller, I don't see any reason to cover this part. If the `dst` is relative path, just let the runtime throw the error. There is no relative path. It's undefined behavior. It might be relative to working dir or rootfs. I didn't see the document for this (If I miss something, please let me know)

Reply:
@fuweid No, sadly there are relative paths (besides unit tests) IIUC. 

If a Dockerfile is using `VOLUME` with a relative path, that is possible, supported and we in fact have an integration test for that: https://github.com/containerd/containerd/blob/bdbcf96267e90d74067fd843960768e164b9b25c/integration/images/volume-copy-up/Dockerfile#L28-L34

That is just passed verbatim to the container config here: https://github.com/containerd/containerd/blob/bdbcf96267e90d74067fd843960768e164b9b25c/pkg/cri/server/container_create.go#L337

If you prefer I can change it there and in the sbserver and probably that is enough too. Is that better for you?

Reply:
@fuweid I changed the PR to only change to an abs path in the volumeMounts() functions (that are called for `VOLUME` directives in the Dockerfile).

You can see the commit, but just in case I paste the diff here (for cri/sbserver, for cri/server it is identical):
```diff
+++ pkg/cri/sbserver/container_create.go
@@ -355,6 +355,9 @@ func (c *criService) volumeMounts(containerRootDir string, criMounts []*runtime.
                }
                volumeID := util.GenerateID()
                src := filepath.Join(containerRootDir, "volumes", volumeID)
+               if !filepath.IsAbs(dst) && goruntime.GOOS != "windows" {
+                       dst = filepath.Join("/", dst)
+               }
                // addOCIBindMounts will create these volumes.
                mounts = append(mounts, &runtime.Mount{
                        ContainerPath:  dst,

```

I hope this addresses your concern. Otherwise, let me know! :)